Formula/pioneer.rb

```diff
@@ -33,8 +34,7 @@ def install
     system "./configure", "--disable-dependency-tracking",
                           "--disable-silent-rules",
                           "--prefix=#{prefix}",
-                          "--with-version=#{version}",
-                          "--with-external-liblua"
```

Comment:
You didn't remove the accompanying `lua` dependency here.

Reply:
Thanks. Did you have a chance to look at why it won't use 5.3 successfully?

Reply:
My understanding is that something like this:
```diff
diff --git a/Formula/lua.rb b/Formula/lua.rb
index dc42d972e2..5576b8929b 100644
--- a/Formula/lua.rb
+++ b/Formula/lua.rb
@@ -31,7 +31,7 @@ class Lua < Formula
     inreplace "src/Makefile" do |s|
       s.gsub! "@LUA_PREFIX@", prefix
       s.remove_make_var! "CC"
-      s.change_make_var! "CFLAGS", "#{ENV.cflags} -DLUA_COMPAT_5_2 $(SYSCFLAGS) $(MYCFLAGS)"
+      s.change_make_var! "CFLAGS", "#{ENV.cflags} -DLUA_COMPAT_5_2 -DLUA_COMPAT_APIINTCASTS $(SYSCFLAGS) $(MYCFLAGS)"
       s.change_make_var! "MYLDFLAGS", ENV.ldflags
     end
 
diff --git a/Formula/pioneer.rb b/Formula/pioneer.rb
index cbc13b5d02..33ebae7fca 100644
--- a/Formula/pioneer.rb
+++ b/Formula/pioneer.rb
@@ -29,12 +29,14 @@ class Pioneer < Formula
   def install
     ENV.cxx11
     ENV["PIONEER_DATA_DIR"] = "#{pkgshare}/data"
+    ENV.append "CFLAGS", "-DLUA_COMPAT_APIINTCASTS"
 
     system "./bootstrap"
     system "./configure", "--disable-dependency-tracking",
                           "--disable-silent-rules",
                           "--prefix=#{prefix}",
-                          "--with-version=#{version}"
+                          "--with-version=#{version}",
+                          "--with-external-liblua"
     system "make", "install"
   end
```
Should have worked, but it doesn't, so 😓. I'd just let this one use the internal lua as you have done, it's not ideal but it seems like the easiest solution with upstream [not even sure](https://github.com/pioneerspacesim/pioneer/issues/3712) which direction to move in.

Reply:
Doesn't even look like that new a deprecation:
```
#if defined(LUA_COMPAT_5_1)     /* { */

/* Incompatibilities from 5.2 -> 5.3 */
#define LUA_COMPAT_MATHLIB
#define LUA_COMPAT_APIINTCASTS

/*
@@ LUA_COMPAT_UNPACK controls the presence of global 'unpack'.
** You can replace it with 'table.unpack'.
*/
#define LUA_COMPAT_UNPACK

/*
@@ LUA_COMPAT_LOADERS controls the presence of table 'package.loaders'.
** You can replace it with 'package.searchers'.
*/
#define LUA_COMPAT_LOADERS

/*
@@ macro 'lua_cpcall' emulates deprecated function lua_cpcall.
** You can call your C function directly (with light C functions).
*/
#define lua_cpcall(L,f,u)  \
        (lua_pushcfunction(L, (f)), \
         lua_pushlightuserdata(L,(u)), \
         lua_pcall(L,1,0,0))


/*
@@ LUA_COMPAT_LOG10 defines the function 'log10' in the math library.
** You can rewrite 'log10(x)' as 'log(x, 10)'.
*/
#define LUA_COMPAT_LOG10

/*
@@ LUA_COMPAT_LOADSTRING defines the function 'loadstring' in the base
** library. You can rewrite 'loadstring(s)' as 'load(s)'.
*/
#define LUA_COMPAT_LOADSTRING

/*
@@ LUA_COMPAT_MAXN defines the function 'maxn' in the table library.
*/
#define LUA_COMPAT_MAXN

/*
@@ The following macros supply trivial compatibility for some
** changes in the API. The macros themselves document how to
** change your code to avoid using them.
*/
#define lua_strlen(L,i)         lua_rawlen(L, (i))

#define lua_objlen(L,i)         lua_rawlen(L, (i))

#define lua_equal(L,idx1,idx2)          lua_compare(L,(idx1),(idx2),LUA_OPEQ)
#define lua_lessthan(L,idx1,idx2)       lua_compare(L,(idx1),(idx2),LUA_OPLT)

/*
@@ LUA_COMPAT_MODULE controls compatibility with previous
** module functions 'module' (Lua) and 'luaL_register' (C).
*/
#define LUA_COMPAT_MODULE

#endif                          /* } */
```

So, yeah, I wouldn't say it was worth poking too much personally. I think you'd be going down a rabbit hole.

Reply:
```
Josephs-MacBook-Pro:~ joe$ brew formula-analytics --install pioneer
install events in the last 30 days for pioneer
=====================================================================================================
1 | pioneer                                                                            | 23 | 100.00%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ 
```
So this one may just get the axe at some point.